swarm/src/handler.rs

```diff
@@ -467,7 +467,7 @@ where
             ConnectionHandlerUpgrErr::Timer => {
                 write!(f, "Timer error while opening a substream")
             }
-            ConnectionHandlerUpgrErr::Upgrade(err) => write!(f, "{err}"),
+            err @ ConnectionHandlerUpgrErr::Upgrade(_) => crate::print_error_chain(f, err),
```

Comment:
I don't think this is what we want, instead, this should probably just print "upgrade error" without printing the inner error. That is what `print_error_chain` then does because the inner error is returned from `source`.

Reply:
If you use `print_error_chain` here, we might end up with duplicate printing if somebody uses `print_error_chain` on `ConnectionHandlerUpgrErr`.